content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+1.  **Create the necessary package infrastructure** when you run:
+
+    ```{r, eval = FALSE}
+    usethis::create_package(getwd())
+    ```
+
+2. The console output will ask if you’d like to overwrite the pre-existing R project. __Select *No*__.
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-overwrite.png")
+    ```
+
+3. A second session of RStudio will open – this session has a *Build* tab in the pane which also has the tabs Environment, History, etc. This tab is specific for building packages, and we will use it later. **You can close the first RStudio instance**.
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-build.png")
+    ```
+
+    * Your project directory should end up looking something like this:
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-directory.png")
+    ```
+
+
+
+### Add files and directories for a tutorial
+
+Now that the package structure is in place, it's time to add tutorial content. Again, the usethis package will do the dirty work for us to create the folders and files we need for a learnr tutorial within our package.
+
+1. **Create the learnr tutorial** and open it for editing when you run `usethis::use_tutorial("<tutorial-name>", "<Tutorial Title>")`:
+
+    ```{r, eval = FALSE}
+    usethis::use_tutorial("lesson1", "Your First Lesson", open = interactive())
+    ```
+
+    * The first argument is the basename for the learnr R Markdown file you are creating. You do not need the `.Rmd` file extension here.
+
+    *  The second argument is what you want the user-facing title of the tutorial to be.
+
+    * This creates an `inst/` directory with some subfolders. If you're not new to creating learnr tutorials, the folder structure here will look familiar to you -- there's nothing different about learnr tutorial directories inside and outside of packages.
+
+      You can open up `inst/` and its subsequent subfolders until you reach the `.Rmd` file you created:
+
+      ```
+      /inst/tutorials/<tutorial-name>/<tutorial-name.Rmd>
+      ```
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-inst.png")
+    ```
+
+1. **Edit the open `.Rmd` file** and customize your tutorial as you normally would.
+
+1. **Repeat** for each additional tutorials you want to include in this package.
+
+
+
+### Build, install, and use your tutorial package
+
+Time to build the package and install it in your R system library.
+
+1. __Navigate to the *Build* pane__ of your RStudio IDE. Then __click *Install and Restart*__. Alternatively, run:
+
+    ```{r, eval = FALSE}
+    devtools::install()
+    ```
+
+    Believe it or not, you've now made a package with a learnr tutorial inside!
```

Comment:
"inside" or "inside it" (latter sounds better to me but I'm ESL and this is one of those things I'm never sure about!)